Replete/ViewController.swift

```diff
@@ -311,8 +311,10 @@ extension ViewController {
               !cmd.isEmpty else { return }
         loadMessage(true, isInput: true, text: cmd)
         ctx.setWidth(self.terminalWidth);
-        ctx.evaluate(cmd)
         inputTextView?.string = "";
+        DispatchQueue.main.async {
+            self.ctx.evaluate(cmd)
+        }
     }
```

Comment:
@wildthink Ahh you are right, there is something wrong with this. Even though it makes things empirically work, it is evidently only avoiding some sort of race. Will investigate further to see if I can create a proper patch for the issue.

Reply:
It appears that some JavaScript objects end up in the auto-release pool, which is drained on the  UI thread. When we go to evaluate a long-running expression, you can end up having the UI block when those JavaScript objects are being cleaned up, as that creates a situation where both the UI and that long running expression are contenting for JavaScriptCore. (You can see this if you use Activity Monitor to sample things).

I suspect the (incorrect) revision in this patch simply avoids the race, and that there is a better way to deal with this.

(The UI is going to ultimately block anyway if you attempt to type after starting a long-running form, because parinfer involves JSC; this ticket is really only about somehow getting the UI to update before the long-running form takes over.)